library/core/src/slice/ascii.rs

```diff
@@ -51,10 +51,28 @@ impl [u8] {
     /// Same as `to_ascii_lowercase(a) == to_ascii_lowercase(b)`,
     /// but without allocating and copying temporaries.
     #[stable(feature = "ascii_methods_on_intrinsics", since = "1.23.0")]
+    #[rustc_const_unstable(feature = "const_eq_ignore_ascii_case", issue = "131719")]
     #[must_use]
     #[inline]
-    pub fn eq_ignore_ascii_case(&self, other: &[u8]) -> bool {
-        self.len() == other.len() && iter::zip(self, other).all(|(a, b)| a.eq_ignore_ascii_case(b))
+    pub const fn eq_ignore_ascii_case(&self, other: &[u8]) -> bool {
+        if self.len() != other.len() {
+            return false;
+        }
+
+        // FIXME(const-hack): Revert this implementation when `core::iter::zip` is allowed in const.
```

Comment:
Revert to what? A few years down the road it can be non-trivial to figure out the history of a comment like this.

Reply:
Understood. I've included the original implementation in the comment.

https://github.com/rust-lang/rust/pull/131721/commits/dedc441fa5e370d8ec4eeb6df8cb13f6ab179e13